In [1]:
import numpy as np
import pandas as pd
import plotly.express as px
import emoji

import os
print(os.listdir("../data"))
import warnings
warnings.filterwarnings('ignore')

['IMDB Dataset.csv']


In [2]:
df=pd.read_csv('../data/IMDB Dataset.csv')
print("Data shape: ", df.shape) # Total review
df

Data shape:  (50000, 2)


review sentiment
0      One of the other reviewers has mentioned that ...  positive
1      A wonderful little production. <br /><br />The...  positive
2      I thought this was a wonderful way to spend ti...  positive
3      Basically there's a family where a little boy ...  negative
4      Petter Mattei's "Love in the Time of Money" is...  positive
...                                                  ...       ...
49995  I thought this movie did a down right good job...  positive
49996  Bad plot, bad dialogue, bad acting, idiotic di...  negative
49997  I am a Catholic taught in parochial elementary...  negative
49998  I'm going to have to disagree with the previou...  negative
49999  No one expects the Star Trek movies to be high...  negative

[50000 rows x 2 columns]

In [3]:
df['sentiment'].value_counts()

sentiment
positive    25000
negative    25000
Name: count, dtype: int64

=> Dataset is balanced

In [4]:
# count word in review
df['review_length'] = df['review'].apply(lambda x:len(str(x).split()))
df['review_length'].head()

0    307
1    162
2    166
3    138
4    230
Name: review_length, dtype: int64

In [5]:
df.describe()

review_length
count   50000.000000
mean      231.156940
std       171.343997
min         4.000000
25%       126.000000
50%       173.000000
75%       280.000000
max      2470.000000

std = 171.34 <br>
The higher this number, the greater the difference between the reviews. <br>

5% (median) = 173 <br>
Typical review length is around 173 words (median) <br>

mean = 231 <br>
Average review normally has 231 words <br>

In [6]:
diff = df['review_length'].max() - df['review_length'].min()
diff

np.int64(2466)

In [7]:
median_review_length = np.median(df['review_length'])
median_review_length

np.float64(173.0)

In [8]:
fig = px.histogram(df, x="review_length",title="Distribution of review length")
fig.show()

Distribution is right-skewed with a long tail of very long reviews

In [9]:
def pick_emojis(text: str):
    """
    Pick emojis from a piece of text
    :param text: string
    :return: emojis
    """
    text = str(text)
    emojis = []

    for char in text:
        if char in emoji.EMOJI_DATA:
            emojis.append(char)

    return "".join(emojis)

In [10]:
df['only_emojis'] = df['review'].apply(pick_emojis)

In [11]:
df['only_emojis'].value_counts()

only_emojis
     49995
®        4
©        1
Name: count, dtype: int64

## Day 1 – Key Takeaways
- Dataset is balanced (50/50)
- Typical review length ~173 words (median)
- Reviews vary significantly in length (std ~171)
- Distribution is right-skewed with long reviews as outliers
- Text noise exists (HTML, emojis)